packages/aws-lambda/layer/bin/publish-layer.sh

```diff
@@ -315,54 +283,53 @@ popd > /dev/null
 export AWS_PAGER=""
 export AWS_MAX_ATTEMPTS=$AWS_CLI_RETRY_MAX_ATTEMPTS
 
-if [[ -z "$BUILD_X86_64_FAIL" ]]; then
-    export BUILD_X86_64_FAIL=0
-fi
-
-if [[ -z "$BUILD_arm64_FAIL" ]]; then
-    export BUILD_arm64_FAIL=0
-fi
-
 if [[ -z $SKIP_AWS_PUBLISH_LAYER ]]; then
   echo "step 6/9: publishing $ZIP_NAME as AWS Lambda layer $LAYER_NAME to specifed regions"
+  
+  if [[ "$PUBLISH_TO_CHINA_REGIONS" == "true" ]]; then
+    echo "Publishing to Chinese regions: $CHINESE_REGIONS"
+    REGIONS="$CHINESE_REGIONS"
+  else
+    # Remove Chinese regions from REGIONS
+    for china_region in $CHINESE_REGIONS; do
+
+      FILTERED_REGIONS=""
+      for region in $REGIONS; do
+        if [[ "$region" != "$china_region" ]]; then
+          FILTERED_REGIONS+="$region "
+        fi
+      done
+      # Update REGIONS with the filtered list
+      REGIONS="$FILTERED_REGIONS"
+    done
+    echo "Publishing to regions except Chinese regions.."
+  fi
+
 
   while read -r region; do
-    skip=0
+    if [[ "$SKIPPED_REGIONS" == *"$region"* ]]; then
+      echo "Skipping region: $region"
+      continue
+    fi
 
-    while read -r skip; do
-      if [[ "$region" == "$skip" ]]; then
-        skip=1
-        break
-      fi
-    done <<< "$SKIPPED_REGIONS"
-
-    if [[ $skip -eq 1 ]]; then
-      echo " - skipping region: $region"
-    else
-      echo " - publishing to region $region:"
-
-      is_chinese_region=0
-      while read -r chinese_region; do
-        if [[ "$region" == "$chinese_region" ]]; then
-          is_chinese_region=1
-          break
-        fi
-      done <<< "$CHINESE_REGIONS"
+    # Publish to AWS, handle Chinese regions
+    echo Publishing AWS Lambda layer in region $region.
 
-      aws_cli_timeout_options="--cli-connect-timeout $AWS_CLI_TIMEOUT_DEFAULT"
+    aws_cli_timeout_options="--cli-connect-timeout $AWS_CLI_TIMEOUT_DEFAULT"
 
-      if [[ $is_chinese_region -eq 1 ]]; then
-        if [[ -z $AWS_ACCESS_KEY_ID_CHINA ]] || [[ -z $AWS_SECRET_ACCESS_KEY_CHINA ]]; then
-          printf "Error: Trying to publish to Chinese region $region, but at least one of the environment variables\n"
-          printf "AWS_ACCESS_KEY_ID_CHINA or AWS_SECRET_ACCESS_KEY_CHINA is not set.\n"
-          exit 1
-        fi
-        # Publishing to a Chinese regions requires different credentials, because it is a different AWS account. The
-        # AWS credential environment variables will be reverted after the publish for this region is done.
-        AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID_CHINA
-        AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY_CHINA
-        aws_cli_timeout_options="--cli-read-timeout $AWS_CLI_TIMEOUT_FOR_CHINA --cli-connect-timeout $AWS_CLI_TIMEOUT_FOR_CHINA"
+    if [[ "$PUBLISH_TO_CHINA_REGIONS" == "true" && "$CHINESE_REGIONS" == *"$region"* ]]; then
```

Comment:
qs: What does this check ""$CHINESE_REGIONS" == *"$region"*"? I assume its checking if chinese_regions is empty? If its empty and PUBLISH_TO_CHINA_REGIONS is true, we should quit the script or?

Reply:
It does not check if `CHINESE_REGIONS` is empty. The condition `"$CHINESE_REGIONS" == *"$region"*" `checks if the value of region matches one of the values listed in CHINESE_REGIONS. 
I have refactored the code.